sam/tag.go

```diff
@@ -0,0 +1,307 @@
+package sam
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"strings"
+)
+
+// QueryTag returns the value of tag t for read s. Currently QueryTag
+// is only supported if the sam was read from a bam file. If no tag
+// data is present, or if the record was not from a bam file, QueryTag
+// will return an error. The input tag t must be exactly 2 characters.
+//
+// The returned value is an interface that may store any of: uint8, int8,
+// uint16, int16, uint32, int32, float32, string, or a slice of interface{}
+// with any of the previous underlying types.
+// It is the callers responsibility to know/determine the type
+// and perform the assertion.
+//
+// The second return is false if the requested tag was not found.
+func QueryTag(s Sam, t string) (value interface{}, found bool, error error) {
+	if s.parsedExtra == nil {
```

Comment:
Just a commenting that the tags should probably be re-used in the giraf package as well